pkg/storage/stores/shipper/compactor/deletion/gennumber_loader_test.go

```diff
@@ -0,0 +1,50 @@
+package deletion
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetCacheGenNumber(t *testing.T) {
+	s := &mockGenNumberGetter{
+		genNumbers: map[string]string{
+			"tenant-a": "1000",
+			"tenant-b": "1050",
+		},
+	}
+	tombstonesLoader := NewGenNumberLoader(s, nil)
```

Comment:
```suggestion
	cacheGenNumberLoader := NewGenNumberLoader(s, nil)
```
This also needs to change later on.